server/service/osquery_utils/queries.go

```diff
@@ -1300,71 +1294,79 @@ func directIngestChromeProfiles(ctx context.Context, logger log.Logger, host *fl
 	return ds.ReplaceHostDeviceMapping(ctx, host.ID, mapping, fleet.DeviceMappingGoogleChromeProfiles)
 }
 
+// directIngestBattery ingests battery data from a host on a Windows or macOS platform
+// and calculates the battery health based on cycle count and capacity.
+// Due to a known osquery issue with M1 Macs (https://github.com/fleetdm/fleet/issues/6763)
+// and the ommission of the `health` column on Windows, we are not leveraging the `health`
+// column in the query and instead aligning the definition of battery health with between
+// macOS and Windows.
 func directIngestBattery(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string) error {
 	mapping := make([]*fleet.HostBattery, 0, len(rows))
+
 	for _, row := range rows {
-		cycleCount, err := strconv.Atoi(EmptyToZero(row["cycle_count"]))
+		health, cycleCount, err := generateBatteryHealth(row, logger)
 		if err != nil {
-			return err
+			level.Error(logger).Log("op", "directIngestBattery", "hostID", host.ID, "err", err)
 		}
 
-		switch host.Platform {
-		case "darwin":
-			mapping = append(mapping, &fleet.HostBattery{
-				HostID:       host.ID,
-				SerialNumber: row["serial_number"],
-				CycleCount:   cycleCount,
-				// database type is VARCHAR(40) and since there isn't a
-				// canonical list of strings we can get for health, we
-				// truncate the value just in case.
-				Health: fmt.Sprintf("%.40s", row["health"]),
-			})
-		case "windows":
-			health, err := generateWindowsBatteryHealth(row["designed_capacity"], row["max_capacity"])
-			if err != nil {
-				level.Error(logger).Log("op", "directIngestBattery", "hostID", host.ID, "err", err)
-			}
-
-			mapping = append(mapping, &fleet.HostBattery{
-				HostID:       host.ID,
-				SerialNumber: row["serial_number"],
-				CycleCount:   cycleCount,
-				Health:       health,
-			})
-		}
+		mapping = append(mapping, &fleet.HostBattery{
+			HostID:       host.ID,
+			SerialNumber: row["serial_number"],
+			CycleCount:   cycleCount,
+			Health:       health,
+		})
 	}
+
 	return ds.ReplaceHostBatteries(ctx, host.ID, mapping)
 }
 
 const (
-	batteryStatusUnknown     = "Unknown"
-	batteryStatusDegraded    = "Check Battery"
-	batteryStatusGood        = "Good"
-	batteryDegradedThreshold = 80
+	batteryStatusUnknown      = "Unknown"
+	batteryStatusDegraded     = "Service recommended"
+	batteryStatusGood         = "Normal"
+	batteryDegradedThreshold  = 80
+	batteryDegradedCycleCount = 1000
 )
 
-func generateWindowsBatteryHealth(designedCapacity, maxCapacity string) (string, error) {
+// generateBatteryHealth calculates the battery health based on the cycle count and capacity.
+func generateBatteryHealth(row map[string]string, logger log.Logger) (string, int, error) {
```

Comment:
Can we scan this data into a struct instead?

Reply:
I ran a quick benchmark on a json scan refactor and was a bit surprised how much overhead was in the json test
  
json: 1,141 ns/op
vs
original:  30.41 ns/op

```
        type BatteryData struct {
		DesignedCapacity string `json:"designed_capacity"`
		MaxCapacity      string `json:"max_capacity"`
		CycleCount       string `json:"cycle_count"`
	}

	rowJSON, err := json.Marshal(row)
	if err != nil {
		return batteryStatusUnknown, 0, fmt.Errorf("failed to marshal row: %v", err)
	}

	var data BatteryData
	if err := json.Unmarshal(rowJSON, &data); err != nil {
		return batteryStatusUnknown, 0, fmt.Errorf("failed to unmarshal row to struct: %v", err)
	}
```

Reply:
Hm, okay. Didn't realize this was dealing with JSON serialization, and that difference seems high given that deserialization is happening either way and the overhead should just be amrshalling. But...fair enough.